packages/service-worker/src/module.ts

```diff
@@ -96,17 +97,27 @@ export function ngswAppInitializer(
     if (typeof options.registrationStrategy === 'function') {
       readyToRegister$ = options.registrationStrategy();
     } else {
-      const [strategy, ...args] = (options.registrationStrategy || 'registerWhenStable').split(':');
+      // Our default is when the app is stable, or after 30 seconds
+      // This ensures that the SW is still registered, in the case of a long running 
+      // or recurring timeout. 
+      const [strategy, ...args] = (options.registrationStrategy || 'registerWhenStable:30000').split(':');
       switch (strategy) {
         case 'registerImmediately':
           readyToRegister$ = of (null);
           break;
         case 'registerWithDelay':
-          readyToRegister$ = of (null).pipe(delay(+args[0] || 0));
+          ngZone.runOutsideAngular(() => {
+            readyToRegister$ = of (null).pipe(delay(+args[0] || 0));
+          });
           break;
         case 'registerWhenStable':
-          const appRef = injector.get<ApplicationRef>(ApplicationRef);
-          readyToRegister$ = appRef.isStable.pipe(filter(stable => stable));
+            ngZone.runOutsideAngular(() => {
+              if (+args[0]) {
+                readyToRegister$ = merge(whenStable(injector), of(null).pipe(delay(+args[0])));
+              } else {
+                readyToRegister$ = whenStable(injector);
+              }
+            });
```

Comment:
Or rather:

```suggestion
          readyToRegister$ = (args[0] === undefined) ?
              whenStable(injector) :
              ngZone.runOutsideAngular(() => merge(whenStable(injector), of(null).pipe(delay(+args[0]))));
```

But, again, we should verify whether `runOutsideAngular()` is needed here or when subscribing to the observable.

Reply:
@gkalpak, you are right. The app will never stabilize. Here is the documentation:

https://angular.io/api/core/ApplicationRef#isstable-examples-and-caveats

> the application will never be stable if you start any kind of recurrent asynchronous task when the application starts (for example for a polling process, started with a setInterval, a setTimeout or using RxJS operators like interval);

By wrapping the code block in the `runOutsideAngular` function, we should be fine.